### PR TITLE
Assemble stdlib.content from snippets

### PR DESCRIPTION
**Pull request by jeffmccune**
Without this patch the stdlib script is a single flat file.  In [1] the
idea of splitting functionality out, to be opted-in using Terraform input
variables has been proposed.

The remaining functionality to be ported to stdlib is somewhat
platform-specific.  Managing sudoers, configuring init scripts, and setting
up gsutil with crcmod.  Pluggable functionality to opt-in to platform
specific behavior should be added ahead of these features so they are not
forced on everyone.

This patch envisions two approaches to opt-in behavior.  Neither approach
is implemented by this patch, only the groundwork to assemble the complete
stdlib content from fragments.

 1. Feature flags such as enable_init_gsutil_crcmod_el for functionality
    maintained directly in the startup-scripts module.
 2. A general purpose `user_content` to serve the use case of
    centralizing project specific functionality.  The intended use case
    is to provide a single place for multiple teams working together to
    add org-wide changes and functionality.  An example of this use case
    is one organization using a `set_dns()` function to manage DNS
    across many teams.  While this could be accomplished with the
    accounts module, the use case remains, to provide a single place to
    make broad changes in behavior.  NOTE: perhaps this use case is
    better served by the module taking in a URL to a GCS bucket and
    sourcing org-specific functionality from that location.

Note, templates have not been used because file fragments accomplish the
same goal, Terraform template behavior is changing soon in version 0.12,
and the `*.sh` files allow the automated shellcheck checks to run
unmodified.

[1]:
https://github.com/terraform-google-modules/terraform-google-startup-scripts/pull/11#issuecomment-456855295

### CODE REVIEW

File: Makefile
Comment (jeffmccune) [inline Makefile:27]: Assembling the `content` output from fragments requires terraform in the docker image.  It was not previously there.

@Jberlinsky What's the process for getting a new docker image in place?

File: test/spec/test_helper/load.bash
Comment (jeffmccune) [inline test/spec/test_helper/load.bash:127]: This slows test runs down, but it's still under 1 minute real time (~30secs with `make docker_bats`, ~20secs with `make docker_bats_parallel`)

File: main.tf
Comment (jeffmccune) [inline main.tf:24]: This is how the fragments are assembled from the feature flags exposed to the user as input vars.

**Review by adrienthebo — COMMENTED** [review]
LGTM so far, will give a formal 👍 once the PR is completed.

File: Makefile
Comment (jeffmccune) [reply Makefile:27]: Nevermind, I see this is handled in `terraform-google-startup-scripts.yml` `make docker_build_bats`

**Comment by morgante** [comment]
> Note, templates have not been used because file fragments accomplish the
same goal, Terraform template behavior is changing soon in version 0.12,
and the *.sh files allow the automated shellcheck checks to run
unmodified.

Just a note that I think we might need templates eventually, but fine with this approach to start.

Overall looks like a good framework.

**Comment by jeffmccune** [comment]
Agreed, I too think we'll likely need templates.  I'm hoping we can find a
use case to build the functionality at the right time.

Thanks for looking into this Morgante.

**Comment by jeffmccune** [comment]
> LGTM so far, will give a formal 👍 once the PR is completed.

@adrienthebo This is completed, please see overall plan I proposed in the bug, "Make startup-scripts pluggable using module input variables"

**Review by adrienthebo — DISMISSED** [review]
LGTM. @morgante you want to check this over before we merge?

File: main.tf
Comment (morgante) [inline main.tf]:
```suggestion
  gsutil_el   = "${var.enable_init_gsutil_crcmod_el ? "" : file("${path.module}/files/init_gsutil_crcmod_el.sh")}"
```

**Review by morgante — CHANGES_REQUESTED** [review]

File: main.tf
Comment (jeffmccune) [reply main.tf]: Fixed in 9ad66f5...cc044ff HEAD -> plugin_mechanism (forced update)

File: main.tf
Comment (jeffmccune) [inline main.tf:19]: @morgante Here's the fix for https://github.com/terraform-google-modules/terraform-google-startup-scripts/pull/12#discussion_r254840934

**Review by morgante — APPROVED** [review]

**Pull request merged**
